src/exercises/course_representation_types.py

```diff
@@ -0,0 +1,99 @@
+from collections import namedtuple
+from pathlib import Path
+from typing import List
+
+
+class Exercice(namedtuple("Exercise", ["name"])):
+    pass
+
+
+class Directory(namedtuple("Directory", ["name", "children"])):
+    def list_exercises(self, prefix) -> List[Path]:
```

Comment:
Problem is
```
 elif isinstance(child, Directory):
                res += child.list_exercises(dir_prefix)
 ``` 
 Does not return a single value that should be stored in the list.
 It return another list of value, that should be append to the list. Does not look like a use case of list comprehension to me. Doesn't they return a list of the same length of the original ?

Reply:
So I would look like this:

```

        return [
            Path(f"{prefix}{dir_prefix}{child.name}.cairo")
            if isinstance(child, Exercice)
            else child.list_exercises(dir_prefix).chain()
            for child in self.children
            if isinstance(child, Exercice) or isinstance(child, Directory)
        ]
```

instead of
```
        for child in self.children:
            if isinstance(child, Exercice):
                res.append(Path(f"{prefix}{dir_prefix}{child.name}.cairo"))
            elif isinstance(child, Directory):
                res += child.list_exercises(dir_prefix)
```

Excuse my non pythonic style, but for the same of readability I will keep my `for if else` codeblock